### DISCUSSION TITLE
How can I materialize assets on aws glue?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18191] [Q&A] [answered]
I am creating a custom step launcher to run asset materialization on AWS Glue. When my asset materialize, launch_step() function is being called. In this function I have the logic to run scripts on glue. In the glue script I am materializing my asset by calling asset_function(). "asset_function()" is my asset definition. This works. However if my asset definition expects a parameter or if its a downstream asset, how can we materialize this from a script which runs on Glue?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700569507967499?thread_ts=1700569507.967499&cid=C01U954MEER).*

**Comment by benpankow** [comment, marked as answer]
Hi Salman, the information you're interested in is available in the step context passed to `launch_step`, which can in turn be turned into an `AssetExecutionContext` (`AssetExecutionContext(step_context)`). You could serialize some or any of the information in this context and pass it to Glue.

This might also be a good use-case for [Dagster Pipes]( https://docs.dagster.io/guides/dagster-pipes), which can handle some of the serialization/passback to external execution environments for you.
